packages/babel-preset-typescript/src/plugin-rewrite-ts-imports.ts

```diff
@@ -16,7 +16,7 @@ export default declare(function ({ types: t }) {
           ? node.importKind
           : node.exportKind;
         if (kind === "value" && source && /[\\/]/.test(source.value)) {
-          source.value = source.value.replace(/(\.[mc]?)ts$/, "$1js");
+          source.value = source.value.replace(/(\.[mc]?)tsx?$/, "$1js");
```

Comment:
Not sure if `.mtsx` or `.ctsx` are valid? [Seems like no](https://github.com/microsoft/TypeScript/issues/44442).

Reply:
They are not valid, but the regexp "accidentally" catching them seems fine.